checkmedia.c

```diff
@@ -1,598 +1,205 @@
-  unsigned size;		/* in kb */
-  unsigned hybrid_size;		/* hybrid partition, in kb */
-  unsigned media_nr;		/* media number */
-
-  char *media_type;		/* media type */
-  char vol_id[33];		/* volume id */
-  char app_id[81];		/* application id */
-  char app_data[0x201];		/* app specific data*/
-  unsigned pad;			/* pad size in kb */
-  unsigned pad_is_skip:1;	/* skip last sectors */
-} iso;
+  unsigned verbose:1;
+  unsigned help:1;
+  unsigned version:1;
+  char *file_name;
+} opt;
+
+struct option options[] = {
+  { "help", 0, NULL, 'h' },
+  { "verbose", 0, NULL, 'v' },
+  { "version", 0, NULL, 1 },
+  { }
+};
 
 
 int main(int argc, char **argv)
 {
   int i;
-  unsigned opt_verbose = 0;
-  unsigned opt_help = 0;
-  unsigned opt_version = 0;
-
-  if(argc > 1) {
-    if(!strcmp(argv[1], "-v") || !strcmp(argv[1], "--verbose")) {
-      opt_verbose = 1;
-      argc--;
-      argv++;
-    }
-    else if(!strcmp(argv[1], "-h") || !strcmp(argv[1], "--help")) {
-      opt_help = 1;
-    }
-    else if(!strcmp(argv[1], "--version")) {
-      opt_version = 1;
-    }
-  }
+  mediacheck_t *media;
 
-  if(opt_help || argc != 2) {
-    printf(
-      "usage: checkmedia [options] iso\n"
-      "Check SUSE installation media.\n"
-      "Options:\n"
-      "  --verbose\tshow some more info\n"
-      "  --version\tdisplay version number\n"
-      "  --help\tshort help text\n"
-    );
+  opterr = 0;
 
-    return 1;
+  while((i = getopt_long(argc, argv, "hv", options, NULL)) != -1) {
+    switch(i) {
+      case 1:
+        opt.version = 1;
+        break;
+
+      case 'v':
+        opt.verbose = 1;
+        break;
+
+      default:
+        help();
+        return i == 'h' ? 0 : 1;
+    }
   }
 
-  if(opt_version) {
+  if(opt.version) {
     printf(VERSION "\n");
-
     return 0;
   }
 
-  get_info(argv[1], opt_verbose);
-
-  if(iso.err) {
-    printf("%s: not an iso\n", argv[1]);
-
-    return 1;
+  if(argc == optind + 1) {
+    opt.file_name = argv[optind];
   }
-
-  if(!iso.digest.type) {
-    printf("%s: no digest found\n", argv[1]);
-
+  else {
+    help();
     return 1;
   }
 
-  printf("    app: %s\n  media: %s%d\n   size: %u kB\n",
-    iso.app_id,
-    iso.media_type,
-    iso.media_nr ?: 1,
-    iso.size
-  );
-
-  if(iso.hybrid_size) printf(" hybrid: %u kB\n", iso.hybrid_size);
-
-  printf("    pad: %u kB\n", iso.pad);
+  media = mediacheck_init(opt.file_name, progress);
 
-  if(iso.pad >= iso.size) {
-    printf("wrong padding value\n");
-    return 1;
+  if(opt.verbose) {
+    for(i = 0; i < sizeof media->tags / sizeof *media->tags; i++) {
+      if(!media->tags[i].key) break;
+      printf("       tags: key = \"%s\", value = \"%s\"\n", media->tags[i].key, media->tags[i].value);
+    }
   }
 
-  if(opt_verbose) {
-    printf("    ref: ");
-    if(iso.digest.got_old) for(i = 0; i < iso.digest.size; i++) printf("%02x", iso.digest.old[i]);
-    printf("\n");
+  if(media->err) {
+    printf("%s: not a supported image format\n", media->file_name);
```

Comment:
should go to STDERR.

Reply:
No (and also the following cases):, all reports go to stdout - it's not a user error, it's a result.